src/main/cli/process.ts

```diff
@@ -0,0 +1,133 @@
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("import <path>",
+            "import epub or lpcl file",
+            (y) =>
+                y.positional("path", {
+                    describe: "absolute, relative or globbing path",
+                    type: "string",
+                })
+                .example("import", "\"./myPath/**/*.epub\"")
+                .example("import", "\"/*.epub\"")
+                .example("import", "\"myPublication.epub\"")
+            ,
+            (argv) => {
+                const pathArray = glob.sync(argv.path, {
+                    absolute: true,
+                    realpath: true,
+                }) || [];
+                const promise = cliImport(pathArray);
+                promise.then((isValid) => {
+                    if (isValid) {
+                        app.exit(0);
+                        return;
+                    }
+                    app.exit(1);
+                }).catch(() => {
+                    app.exit(1);
+                });
+            },
+        )
+        .command("read <title>",
+            "searches already-imported publications with the provided TITLE, and opens the reader with the first match",
+            (y) =>
+                y.positional("title", {
+                    describe: "title of your publication",
+                    type: "string",
+                })
+            ,
+            (argv) => {
+                mainFct();
+                app.on("ready", async () => {
+                    try {
+                        if (!await cliRead(argv.title)) {
+                            debug("no publication to read");
+                        }
+                    } catch (e) {
+                        debug("An error occurred during cliRead");
+                    }
+                });
+            },
+        )
+        .command("$0 [path]",
+            "default command",
+            (y) =>
+                y.positional("path", {
+                    describe: "path of your publication, it can be an absolute, relative path",
+                    type: "string",
+                    coerce: (arg) => path.resolve(arg),
+                })
+                .completion()
+            ,
+            (argv) => {
+                mainFct();
+                if (argv.path) {
+                    app.on("ready", async () => {
+                        try {
+                            if (!await cli_(argv.path)) {
+                                const errorTitle = "Import Failed";
+                                const errorMessage = `Import failed for the publication path : ${argv.path}`;
+                                debug(errorMessage);
+                                dialog.showErrorBox(errorTitle, errorMessage);
+                            }
+                        } catch (e) {
```

Comment:
Suggestion: `process.stderr.write()` and `dialog.showErrorBox()` of the caught exception `e`.

This exception can (for example) be the *uncaught* promise rejection from `const publication = await catalogService.importFile(filePath);` which is called inside the `cli_()` function. So it is important to *not* swallow the context of the error, and to report it appropriately. Do you think `stderr` is necessary as well as `dialog.showErrorBox()`?

Reply:
done